### PR TITLE
Feature/ep 1881

### PR DESCRIPTION
**Pull request by mfranceschit**
## Issue tracker links

[EP-1881](https://autoclouddev.atlassian.net/browse/EP-1881)

## Changes/solution

- Replaced cuid using generateUniqueId util.

## Dependencies

AWS SDK https://github.com/cloudgraphdev/sdk/pull/14

### CODE REVIEW

**Comment by tyler-dunkel** [comment]
@m-pizarro can you do some through testing on this?

File: src/services/securityGroup/format.ts [hunk 1]
```diff
@@ -72,7 +73,12 @@ export default ({
       ;(rule.ipRanges || []).map(
         ({ cidrIp, description: ipRangesDescription = '' }) => {
           allRules.push({
-            id: cuid(),
+            id: generateUniqueId({
+              arn,
+              cidrIp,
+              ipRangesDescription,
```
Comment (m-pizarro) [inline src/services/securityGroup/format.ts:79]: I'm seeing some duplicate id errors:

image

It seems that `cidrIp `and `ipRangesDescription` can be the same between the rules, which would generate the same id in those cases.

File: src/services/securityGroup/format.ts [hunk 2]
```diff
@@ -81,7 +87,12 @@ export default ({
       ;(rule.ipv6Ranges || []).map(
         ({ cidrIpv6, description: ipv6RangesDescription = '' }) => {
           allRules.push({
-            id: cuid(),
+            id: generateUniqueId({
+              arn,
+              cidrIpv6,
+              ipv6RangesDescription,
```
Comment (m-pizarro) [inline src/services/securityGroup/format.ts:93]: same here

**Comment by m-pizarro** [comment]
In addition `addawsSecurityGroup `this error is occurs in `addawsAccount ` and `addawsTag`:

image

image

File: src/services/securityGroup/format.ts (hunk 1 above)
Comment (tyler-dunkel) [reply src/services/securityGroup/format.ts:79]: hmm I wonder why these are happening, unless the same rule is defined twice on the same arn 🤔

Comment (m-pizarro) [reply src/services/securityGroup/format.ts:79]: Not exactly the same rule, but yes those 2 values that compose the Id. For example, the rules of this security group:

image

Comment (tyler-dunkel) [reply src/services/securityGroup/format.ts:79]: hmm maybe we need to change up this obj structure in that case

Comment (m-pizarro) [reply src/services/securityGroup/format.ts:79]: Yes, it should at least include all the properties that make the rule unique. On the other hand here I added a log to the generation of these ids to validate that this is happening:

image

image

Comment (mfranceschit) [reply src/services/securityGroup/format.ts:81]: Hey @m-pizarro I improved the uniqueness of these nested objects. Please give it another try when you have a chance

File: src/services/athenaDataCatalog/format.ts [hunk 3]
```diff
@@ -10,10 +10,12 @@ import { athenaDataCatalogArn } from '../../utils/generateArns'
  */
 const formatColumns = (column: Column) => {
   return {
-    id: cuid(),
+    id: generateUniqueId({
+      ...column,
```
[the remark this answers is not in the document]
Comment (mfranceschit) [reply src/services/athenaDataCatalog/format.ts]: I fixed it. It is preferable that all nested objects includes the arn of the parent to avoid duplicity.

**Comment by mfranceschit** [comment]
> In addition `addawsSecurityGroup `this error is occurs in `addawsAccount ` and `addawsTag`:
> 
> image
> 
> image

I haven't seen these errors after I fixed the duplication problems for SG.

File: src/services/securityGroup/format.ts (hunk 1 above)
Comment (m-pizarro) [reply src/services/securityGroup/format.ts:81]: @mfranceschit yeah, everything looks good now, no more errors are logged. Thanks!

**Pull request merged**